### PR TITLE
fix: note insert lookup on iOS

### PR DESCRIPTION
**Pull request by AhmedNMahran**
- Fixes note insert/lookup edge case on iOS persistence path.
- Touches only note repository/schema/tests.
- Verification: `:persistence:jvmTest :syncengine:jvmTest :sync-pipelines:jvmTest`

### CODE REVIEW

File: persistence/src/commonMain/kotlin/com/quran/shared/persistence/repository/note/repository/NotesRepositoryImpl.kt [hunk 1]
```diff
@@ -56,7 +56,11 @@ class NotesRepositoryImpl(
                 start_ayah_id = startAyahId,
                 end_ayah_id = endAyahId
             )
-            val record = notesQueries.value.getLastInsertedNote()
+            val record = notesQueries.value.getNoteByContent(
```
Comment (ahmedre) [inline persistence/src/commonMain/kotlin/com/quran/shared/persistence/repository/note/repository/NotesRepositoryImpl.kt]: why not just rely on the id instead - this is risky and asking for trouble (encoding issues, etc). can we just use an id?

Comment (AhmedNMahran) [reply persistence/src/commonMain/kotlin/com/quran/shared/persistence/repository/note/repository/NotesRepositoryImpl.kt]: done

**Review by ahmedre — APPROVED** [review]
